src/internal/m365/collection/drive/collections.go

```diff
@@ -982,11 +990,12 @@ func (c *Collections) processItem(
 			return clues.New("item seen before parent folder").WithClues(ictx)
 		}
 
+
 		// This will only kick in if the file was moved multiple times
 		// within a single delta query.  We delete the file from the previous
 		// collection so that it doesn't appear in two places.
-		prevParentContainerID, ok := currPrevPaths[itemID]
-		if ok {
+		prevParentContainerID, prevAdded := currPrevPaths[itemID]
```

Comment:
nit: `prev` is an overload term for us; would be cleaner to avoid it.
```suggestion
		prevParentContainerID, alreadyAdded := currPrevPaths[itemID]
```